src/Claims.ts

```diff
@@ -230,6 +234,103 @@ export class Claims {
     };
   }
 
+  /**
+   * Retrieve a list of Identities with claims associated to them. Can be filtered using parameters
+   *
+   * @param opts.targets - Identities (or Identity IDs) for which to fetch claims (targets). Defaults to all targets
+   * @param opts.trustedClaimIssuers - Identity IDs of claim issuers. Defaults to all claim issuers
+   * @param opts.scope - scope of the claims to fetch. Defaults to any scope
+   * @param opts.claimTypes - types of the claims to fetch. Defaults to any type
+   * @param opts.includeExpired - whether to include expired claims. Defaults to true
+   * @param opts.size - page size
+   * @param opts.start - page offset
+   *
+   * @note supports pagination
+   * @note uses the middleware V2
+   */
+  public async getIdentitiesWithClaimsV2(
+    opts: {
+      targets?: (string | Identity)[];
+      trustedClaimIssuers?: (string | Identity)[];
+      scope?: Scope;
+      claimTypes?: Exclude<ClaimType, ClaimType.InvestorUniquenessV2>[];
+      includeExpired?: boolean;
+      size?: BigNumber;
+      start?: BigNumber;
+    } = {}
+  ): Promise<ResultSet<IdentityWithClaims>> {
+    const { context } = this;
+
+    const {
+      targets,
+      trustedClaimIssuers,
+      scope,
+      claimTypes,
+      includeExpired = true,
+      size = new BigNumber(DEFAULT_GQL_PAGE_SIZE),
+      start = new BigNumber(0),
+    } = opts;
+
+    let targetIssuers;
+
+    const filters = {
+      scope: scope ? scopeToMiddlewareScope(scope) : undefined,
+      trustedClaimIssuers: trustedClaimIssuers?.map(trustedClaimIssuer =>
+        signerToString(trustedClaimIssuer)
+      ),
+      claimTypes: claimTypes?.map(ct => ClaimTypeEnum[ct]),
+      includeExpired,
+    };
+
+    if (!targets) {
+      const targetResults = await context.queryMiddlewareV2<Ensured<QueryV2, 'claims'>>(
+        claimTargets({
+          ...filters,
+        })
+      );
+
+      const {
+        data: { claims },
+      } = targetResults;
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const groupedTargets = claims!.groupedAggregates!;
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      targetIssuers = flatten(groupedTargets.map(groupedAggregate => groupedAggregate.keys!));
+    } else {
+      targetIssuers = targets.map(target => signerToString(target));
+    }
+
+    // note: pagination count is based on the target issuers and not the claims count
+    const count = new BigNumber(targetIssuers.length);
+
+    targetIssuers = targetIssuers.sort().slice(start.toNumber(), size.plus(start).toNumber());
```

Comment:
Tooling gql does pagination based on sorted target issuers. Hence the explicit `sort()` function (as graphql doesn't sort the final data)

Reply:
This belongs in a comment